xarray/core/variable.py

```diff
@@ -1993,8 +1995,20 @@ def quantile(
 
             method = interpolation
 
+        using_numbagg = False
         if skipna or (skipna is None and self.dtype.kind in "cfO"):
-            _quantile_func = np.nanquantile
+            if (
```

Comment:
Sorry for the late comment. 

Would you mind moving this switching logic to `nputils.py` and following this approach please?
https://github.com/pydata/xarray/blob/c9ba2be2690564594a89eb93fb5d5c4ae7a9253c/xarray/core/nputils.py#L280 .

It'd be nice to keep it contained to that file only.

Reply:
Done with https://github.com/pydata/xarray/pull/8684/commits/cbd4ed729b04d089c5df1cbcae1f4ed1e204c023. 
Not sure about the new failing tests. They seem to only occur with methods `'midpoint'` or `'lower'` and python <3.12